src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -0,0 +1,243 @@
+    const double suggested_value = 2.0 * length_inner_cube_ * sqrt(3.0);
+    PARSE_ERROR(
+        context,
+        "The radius for the enveloping cube is too small! The Frustums will be "
+        "malformed. A recommended radius is:\n"
+            << suggested_value);
+  }
+  if (outer_radius_objectA < inner_radius_objectA) {
+    PARSE_ERROR(context,
+                "ObjectA's inner radius must be less than its outer radius.");
+  }
+  if (outer_radius_objectB < inner_radius_objectB) {
+    PARSE_ERROR(context,
+                "ObjectB's inner radius must be less than its outer radius.");
+  }
+  if (use_projective_map_) {
+    projective_scale_factor_ = length_inner_cube_ / length_outer_cube_;
+  } else {
+    projective_scale_factor_ = 1.0;
+  }
+}
+
+Domain<3> BinaryCompactObject::create_domain() const noexcept {
+  const double inner_sphericity_A = excise_interiorA_ ? 1.0 : 0.0;
+  const double inner_sphericity_B = excise_interiorB_ ? 1.0 : 0.0;
+
+  using Maps = std::vector<
+      std::unique_ptr<CoordinateMapBase<Frame::Logical, Frame::Inertial, 3>>>;
+
+  Maps maps;
+  // ObjectA/B is on the left/right, respectively.
+  Maps maps_center_A = wedge_coordinate_maps<Frame::Inertial>(
+      inner_radius_objectA_, outer_radius_objectA_, inner_sphericity_A, 1.0,
+      use_equiangular_map_, xcoord_objectA_, false);
+  Maps maps_cube_A = wedge_coordinate_maps<Frame::Inertial>(
+      outer_radius_objectA_, sqrt(3.0) * 0.5 * length_inner_cube_, 1.0, 0.0,
+      use_equiangular_map_, xcoord_objectA_, false);
+  Maps maps_center_B = wedge_coordinate_maps<Frame::Inertial>(
+      inner_radius_objectB_, outer_radius_objectB_, inner_sphericity_B, 1.0,
+      use_equiangular_map_, xcoord_objectB_, false);
+  Maps maps_cube_B = wedge_coordinate_maps<Frame::Inertial>(
+      outer_radius_objectB_, sqrt(3.0) * 0.5 * length_inner_cube_, 1.0, 0.0,
+      use_equiangular_map_, xcoord_objectB_, false);
+  Maps maps_frustums = frustum_coordinate_maps<Frame::Inertial>(
+      length_inner_cube_, length_outer_cube_, use_equiangular_map_,
+      {{-translation_, 0.0, 0.0}}, projective_scale_factor_);
+  Maps maps_outer_shell = wedge_coordinate_maps<Frame::Inertial>(
+      sqrt(3.0) * 0.5 * length_outer_cube_, radius_enveloping_sphere_, 0.0, 1.0,
+      use_equiangular_map_, 0.0, true);
+
+  std::move(maps_center_A.begin(), maps_center_A.end(),
+            std::back_inserter(maps));
+  std::move(maps_cube_A.begin(), maps_cube_A.end(),
+            std::back_inserter(maps));
+  std::move(maps_center_B.begin(), maps_center_B.end(),
+            std::back_inserter(maps));
+  std::move(maps_cube_B.begin(), maps_cube_B.end(),
+            std::back_inserter(maps));
+  std::move(maps_frustums.begin(), maps_frustums.end(),
+            std::back_inserter(maps));
+  std::move(maps_outer_shell.begin(), maps_outer_shell.end(),
+            std::back_inserter(maps));
+
+
+  //Set up the maps for the central cubes, if any exist.
```

Comment:
[nitpick] whitespace after `//` ... doesn't clang-format do this? Have you run run clang-format on this file?